ride-share.rb

```diff
@@ -0,0 +1,235 @@
+drivers = [
+  {
+    id: "DR0001",
+    rides: [
+      {
+        date: "3rd Feb 2016",
+        rider_id: "RD0003",
+        cost: 10,
+        rate: 3,
+      },
+      {
+        date: "3rd Feb 2016",
+        rider_id: "RD0015",
+        cost: 30,
+        rate: 4,
+      },
+      {
+        date: "5th Feb 2016",
+        rider_id: "RD0003",
+        cost: 45,
+        rate: 2,
+      },
+    ],
+  },
+  {
+    id: "DR0002",
+    rides: [
+      {
+        date: "3rd Feb 2016",
+        rider_id: "RD0073",
+        cost: 25,
+        rate: 5,
+
+      },
+      {
+        date: "4th Feb 2016",
+        rider_id: "RD0013",
+        cost: 15,
+        rate: 1,
+      },
+      {
+        date: "5th Feb 2016",
+        rider_id: "RD0066",
+        cost: 35,
+        rate: 3,
+
+      },
+    ],
+  },
+  {
+    id: "DR0003",
+    rides: [
+      {
+        date: "4th Feb 2016",
+        rider_id: "RD0066",
+        cost: 5,
+        rate: 5,
+      },
+      {
+        date: "5th Feb 2016",
+        rider_id: "RD0003",
+        cost: 50,
+        rate: 2,
+      },
+    ],
+  },
+  {
+    id: "DR0004",
+    rides: [
+      {
+        date: "3rd Feb 2016",
+        rider_id: "RD0022",
+        cost: 5,
+        rate: 5,
+      },
+      {
+        date: "4th Feb 2016",
+        rider_id: "RD0022",
+        cost: 10,
+        rate: 4,
+      },
+      {
+        date: "5th Feb 2016",
+        rider_id: "RD0073",
+        cost: 20,
+        rate: 5,
+      },
+    ],
+  },
+]
+
+# 1 The number of rides each driver has given
+def total_rides(drivers)
+  drivers.each do |driver|
+    rides = driver[:rides].length
+    puts "Driver ID:#{driver[:id]} has given #{rides} rides."
+  end
+end
+
+# 2 The total amount of money each driver has made
+def total_money_made(drivers)
+  drivers.each do |driver|
+    total_cost = 0
+    driver[:rides].each do |ride_info|
+      total_cost += ride_info[:cost]
+    end
+    puts "The total amount of money driver ID:#{driver[:id]} has made is $#{total_cost}."
+  end
+end
+
+# 3 The average rating for each driver
+def average_rating(drivers)
+  drivers.each do |driver|
+    total_rating = 0
```

Comment:
Ratings need to use [floating point math](https://en.wikipedia.org/wiki/Floating-point_arithmetic) since we're planning to get an average and want things like 4.67 as an answer.

Because of the way Ruby converts things if you have one floating point value everything winds up floating point so if you start with `0.0` you should get the correct math.  (Though note that with floating point math you will want to `round` before you display things.)

```suggestion
    total_rating = 0.0
```